blood/src/main/java/cz/cuni/mff/d3s/blood/report/Report.java

```diff
@@ -0,0 +1,59 @@
+package cz.cuni.mff.d3s.blood.report;
+
+import cz.cuni.mff.d3s.blood.report.dump.DumpRegistration;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class Report {
+    private static Report instance = null;
+    private static System.Logger LOGGER = System.getLogger(Report.class.getName());
+    List<DumpRegistration> registrations = new ArrayList<>();
+
+    private Report() {
+        Runtime.getRuntime().addShutdownHook(new Thread(this::onShutdown, "Report shutdown hook"));
+    }
+
+    public static Report getInstance() {
+        if (instance == null)
+            instance = new Report();
+
+        return instance;
+    }
+
+    public void registerDump(DumpRegistration registration) {
+        registrations.add(registration);
+    }
+
+    private void onShutdown() {
+        var startTime = Instant.now();
+
+        new File(DumpHelpers.DUMPS_DIR_NAME).mkdir();
+
+        String timestamp = DumpHelpers.getDateString();
+        File reportDir = new File(DumpHelpers.DUMPS_DIR_NAME + "/report-" + timestamp);
```

Comment:
I find the approach used in `DumpHelpers::getDumpFileWriter` (i.e. `new File(File parent, String child)`) more elegant than concatenating with a string containing directory separator.

Reply:
You are right. Might be even better to look at new Path API...